ntp.go

```diff
@@ -173,27 +189,49 @@ func Query(host string, version int) (*Response, error) {
 }
 
 // getTime returns the "receive time" from the remote NTP server host.
-func getTime(host string, version int) (*msg, error) {
-	if version < 2 || version > 4 {
+func getTime(host string, opt QOption) (*msg, error) {
+	if opt.Version == 0 {
+		opt.Version = defaultNtpVersion
+	}
+
+	if opt.Version < 2 || opt.Version > 4 {
 		panic("ntp: invalid version number")
 	}
 
+	if opt.Timeout == 0 {
+		opt.Timeout = defaultTimeout
+	}
+
 	raddr, err := net.ResolveUDPAddr("udp", host+":123")
 	if err != nil {
 		return nil, err
 	}
 
+	if opt.Port != 0 {
+		raddr.Port = opt.Port
+	}
+
 	con, err := net.DialUDP("udp", nil, raddr)
 	if err != nil {
 		return nil, err
 	}
 	defer con.Close()
-	con.SetDeadline(time.Now().Add(timeout))
+
+	if opt.IpTTL != 0 {
+		ipcon := ipv4.NewConn(con)
+		err = ipcon.SetTTL(opt.IpTTL)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	con.SetDeadline(time.Now().Add(opt.Timeout))
 
 	m := new(msg)
 	m.setMode(client)
-	m.setVersion(version)
-	m.TransmitTime = toNtpTime(time.Now())
+	m.setVersion(opt.Version)
+	xmt := toNtpTime(time.Now())
```

Comment:
`m.OriginTime` becomes echo of `xmt` in incoming packet, but `m.TransmitTime` is overwritten with server timestamp of outgoing packet, so it's not equal `m.OriginTime`.